build/setup-kubectl.sh

```diff
@@ -1,31 +1,33 @@
 #!/bin/bash
 set -e
 
+# Default values for kubectl version, install location, and architecture
+DEFAULT_KUBECTL_VERSION="v1.27.3"
 DEFAULT_INSTALL_LOCATION="/app/kubectl/linux"
-DEFAULT_KUBECTL_VERSION="v1.21.2"
+DEFAULT_ARCH="amd64"
 
-# Arguments: kubectl version, install location
-function installKubectl {
-    local loc_kube_version=$1
-    local loc_setup_location=$2
+# Function to install kubectl
+# Arguments: kubectl version, install location, architecture
+function install_kubectl {
+    # Set default values if arguments are not provided
+    kubectl_version="${1:-$DEFAULT_KUBECTL_VERSION}"
+    install_location="${2:-$DEFAULT_INSTALL_LOCATION}"
+    arch="${3:-$DEFAULT_ARCH}"
 
-    if [ -z "$loc_kube_version" ]; then
-        loc_kube_version=$DEFAULT_KUBECTL_VERSION
+    # Map arm64v8 architecture to arm64
+    if [ "$arch" == "arm64v8" ]; then
+        echo "setting arch to arm64"
+        arch="arm64"
     fi
 
-    if [ -z "$loc_setup_location" ]; then
-        loc_setup_location=$DEFAULT_INSTALL_LOCATION
-    fi
-
-    echo "Setting up kubectl $loc_kube_version in $loc_setup_location"
-    curl -LO https://storage.googleapis.com/kubernetes-release/release/${loc_kube_version}/bin/linux/amd64/kubectl
+    echo "Setting up kubectl $kubectl_version in $install_location with arch $arch"
+    curl -LO "https://storage.googleapis.com/kubernetes-release/release/$kubectl_version/bin/linux/$arch/kubectl"
```

Comment:
I see the pipeline has changed the kubectl download location to something under `https://dl.k8s.io`. Should that be changed here too?

Reply:
Thanks @peterbom I will update them.